<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script>
    /*--------------------------------------------------------------------------------------------------------------------------
      问：程序员自己如何抛出一个自定义的错误对象?
      答：throw new Error(错误信息)
    */
   /*
      问：自定义抛出的错误对象是什么具体的错误类型?
      答：为 error 类型，该类型为 js 引擎抛出的所有错误类型的父类型
    */
    /* function demo(params) {
      if (params % 2 === 0) {
        console.log('输入的数字为偶数')
      } else {
        throw new Error('输入的number为基数，在我们这个程序里被认为是错误的')
      }
    }
    try {
      demo(Date.now())
    } catch (error) {
      console.log(error)
    } */
    // --------------------------------------------------------------------------------------------------------------------------


    /*--------------------------------------------------------------------------------------------------------------------------
    问：throw 关键字 只能抛出错误对象吗？
    答：不是的，throw 可以抛出任何错误（字符串，数字，函数...)
        throw 抛出的任何东西，都会被视为一个未捕获的错误信息（不一定是一个错位对象）
        并且会阻断程序的继续往下执行
    */
    /* throw function() {}
    console.log('wwwwww') */
    //---------------------------------------------------------------------------------------------------------------------------
  
    //==========================================================================================================================================
    /*
    注意🐷：throw xxx 不能被 return, 不然 js 会抛出未捕获的语法错误提示【Uncaught SyntaxError】
    */
   /* (function throwError() {
    return throw 123 // Uncaught SyntaxError
   })() */
    //==========================================================================================================================================
    
    
    
  </script>
</body>
</html>